core-common/src/main/java/org/zalando/nakadi/validation/ISO4217CurrencyValidator.java

```diff
@@ -0,0 +1,40 @@
+package org.zalando.nakadi.validation;
+
+import org.everit.json.schema.FormatValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Currency;
+import java.util.Optional;
+
+public class ISO4217CurrencyValidator implements FormatValidator {
+    private static final Logger LOG = LoggerFactory.getLogger(ISO4217CurrencyValidator.class);
+    private final String eventTypeName;
+    private final String formatName;
+
+    ISO4217CurrencyValidator(final String eventTypeName) {
+        this(eventTypeName, "iso-4217");
+    }
+
+    ISO4217CurrencyValidator(final String eventTypeName, final String formatName) {
```

Comment:
why do we need to pass format name when we have named validator as `ISO4217`CurrencyValidator. Lets keep it in the name of the class only.

Reply:
Because there are at least two format names that seen in practice: `iso-4217` and `ISO-4217`. Hence I'm registering both of them (the `org.everit.json` lib, used for schema validation in the project is case sensitive for format names).

I'm open where to keep the `ISO-4217` constant string.
However, this is not the "approved" format name according to https://opensource.zalando.com/restful-api-guidelines/#238. For now, I hope this is rather a transitional period to have one validator under different format names.

In the future, either Nakadi should normalize the format names (transform to lower case) under the hood (when creating validation objects per event type schema), or alternatively, this can be fixed at data level (by updating event types schemas in the schema registry).
The system also could be more stricter to what it accepts as format names (i.e., validate more stricter the event type schemas).